lib/libsysdecode/utrace.c

```diff
@@ -87,6 +87,29 @@ print_utrace_mrs(FILE *fp, void *p)
 	}
 	return (1);
 }
+
+static int
+print_utrace_c18n(FILE *fp, void *p)
+{
+	struct utrace_c18n *ut = p;
+
+	switch (ut->event) {
+	case UTRACE_COMPARTMENT_ENTER:
+		fprintf(fp, "RTLD: c18n: enter %s from %s at [%zu] %s (%02hhx)",
+		    ut->callee, ut->caller, ut->symnum, ut->symbol, ut->fsig);
+		break;
+	case UTRACE_COMPARTMENT_LEAVE:
+		fprintf(fp, "RTLD: c18n: leave %s to %s at [%zu] %s (%02hhx)",
+		    ut->callee, ut->caller, ut->symnum, ut->symbol, ut->fsig);
+		break;
+	default:
+		return (0);
+	}
+	if (ut->verbose)
```

Comment:
It's a bit unusual to have this flag be set when the trace is logged rather than being something you can choose at inspection time (that is a flag passed in from kdump/truss) since you have the data in the utrace log always.  You can always adjust that later though.  (That is, I think ideally you would have verbose or not come from a flag passed to kdump or truss, or even if this is a temporary thing a hacky thing would be to check an env var here to determine verbose vs non-verbose rather than having it be a property of when you are running the program since you might decide later when examining a dump via kdump that you'd like to see the extra detail for some reason.)